syncmaster/worker/handlers/db/base.py

```diff
@@ -44,6 +44,23 @@ def read(self) -> DataFrame:
                 name=hwm_name,
                 expression=self.transfer_dto.strategy.increment_by,
             )
+            if self.transfer_dto.resources.max_parallel_tasks > 1:
+                reader_params["options"] = {
+                    "partition_column": self.transfer_dto.strategy.increment_by,
+                    "num_partitions": self.transfer_dto.resources.max_parallel_tasks,
+                    "partitioning_mode": "range" if self.hwm and self.hwm.value else "hash",
+                }
```

Comment:
BTW, these options are used only by JDBC sources, not by Hive. Maybe we need dedicated method `def _get_reading_options(self)` which can be overridden by different implementations.